arch/z16/src/common/z16_stackdump.c

```diff
@@ -68,8 +69,12 @@ static void z16_stackdump(void)
       stack = stack_base;
     }
 
+  /* Flush any buffered SYSLOG data to avoid overwrite */
+
+  syslog_flush();
+
   for (stack = stack & ~0x0f;
-       stack < stack_base + stack_size;
+       stack < ((stack_base + stack_size) & ~0x1f);
```

Comment:
```suggestion
       stack < ((stack_base + stack_size) & ~0x0f);
```
or for better compatibility:
```
  for (stack = stack & ~(8 * sizeof(chipreg_t) - 1);
       stack < ((stack_base + stack_size) & ~(8 * sizeof(chipreg_t) - 1));
       stack += 8 * sizeof(chipreg_t))
```